### DISCUSSION TITLE
Can't get rpm display for ePump.

### DISCUSSION THREAD

**Discussion by rpourzia** [aqualinkd/AqualinkD#214] [Q&A] [answered]
My equipment consists of a Jandy ePump ((VSSHP270DV2A), iQ30-RS, iAqualink 3.0, and an RS8 panel.
The logger shows this:

![image](https://github.com/sfeakes/AqualinkD/assets/3403252/c89e9db9-72e8-463a-b4f6-93c359131325)

Here'all the changed lines in my aqualinkd.conf file
`log_file=/var/log/aqualinkd.log
log_level=INFO

ePump = Jandy ePump or ePump AC
read_RS485_ePump = yes
 
button_01_pumpIndex=1
button_01_label=Filter Pump
button_01_pumpID=0x78`

the only other change I have is in the webconfig.js file I made this change:
` var show_vsp_gpm=true;`

As additional information. iAqualink shows this as number of rs485 devices
`Reported RS485 Devices:
ePump 1
Control Panel 1
iAquaLink WiFi 4`
when aqualinkd runs, it shows this:
`Reported RS485 Devices:
ePump 1
Control Panel 1
iAquaLink WiFi 4
AquaLink Touch 2
Serial Adaptr 1
Control Panel 3`
I don't know the address of these devices like 4 for the wifi.

The web interface seems to work and I can turn the pump on and off from the panel but no information about the rpm or power usage is displayed.  The only thing I am not 1100 sure is pumpindex=1.  I know it is the first device in the panel but I am not sure how to verify the index aqualinkd is looking for. 
I would appreciate any ideas.

**Comment by sfeakes** [comment]
The other thing that you need to enable for VSP is `extended_device_id` but it looks like you probably have that looking at what devices are connected in your iAqualink output.
The pump index number is the pump number that's shown in OneTouch or Aqualink Touch interface ie `Jandy ePump 1` displayed would be pump index 1.

Jandy ePumps don't display / use GPM, they use RPM.  So your setting in the `webconfig.js` is incorrect, that might cause this problem as the GPM will never be posted for an ePUMP.  Well that's to my knowledge.

If that doesn't fix it, try using `0x40` for your extended device ID.   That will force AqualinkD to use a different protocol to get the VSP information.  It looks like you followed using one of the preferred ID's, so using 0x40 is a test not permeant solution.

**Comment by rpourzia** [comment]
Setting the `extended_device_id=0x40` or `extended_device_id=0x41`  work! Thank you. The Web UI shows the RPM and I am able to get the wattage from mqtt. So I am happy. Not sure if there are plans for allowing to change the speed at any time in the future.  The iAqualink web interface is so bad.  I would rather do everything with aqualinkd.

**Reply by sfeakes** [reply, marked as answer]
You should be able to change the speed by long press on what ever icon has the rpm showing. Probably filter pump on your case. That should display popup where you can set speed by RPM.
Interesting that it didn’t work in the Aqualink touch protocol, but did with one touch protocol.  I’ll have to check code for a bug/problem.

**Comment by rpourzia** [comment]
I don't know why I did not see this bfore but in the iAqualink 3.0 help menu, I just saw this:
![image](https://github.com/sfeakes/AqualinkD/assets/3403252/003919c1-1fac-417e-9586-eff908b17544)
That explains why the `extended_device_id` values for Aqualink Touch  (0x40, or 0x41) worked in the configuration  even though I don't have a physical Aqualink touch.
